### PR TITLE
bluetooth: host: Disallow directed adv options on undirected adv

### PR DESCRIPTION
**Pull request by rugeGerritsen**
Previously, if the peer address was not set, the host would
do undirected advertising even if the application instructed
the stack to do directed advertising.

Adding this additional parameter validation reduces the confusion
of application developers when they have configured something wrong.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -7946,6 +7946,12 @@ static bool valid_adv_ext_param(const struct bt_le_adv_param *param)
 		}
 	}
 
+	if ((param->options & (BT_LE_ADV_OPT_DIR_MODE_LOW_DUTY |
+			       BT_LE_ADV_OPT_DIR_ADDR_RPA)) &&
```
Comment (balaji-nordic) [inline subsys/bluetooth/host/hci_core.c:7950]: Why is it necessary to also check BT_LE_ADV_OPT_DIR_ADDR_RPA?

Comment (rugeGerritsen) [reply subsys/bluetooth/host/hci_core.c:7950]: When the application selects BT_LE_ADV_OPT_DIR_ADDR_RPA, the application intends to perform directed advertising. I would expect the zephyr host to return an error when the peer address is not specified.

**Review by balaji-srin — APPROVED** [review]

**Comment by joerchan** [comment]
@rugeGerritsen The changes look good, but the commit message is a bit misleading. The BT_LE_ADV_OPT_DIR_MODE_LOW_DUTY is not an instruction to do directed advertising. It is just an option that can be used with directed advertising to set Low Duty directed advertising.

Could you rephrase the commit title and message?
"Disallow directed advertising options on undirected advertising" maybe?

**Review by joerchan — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
